### ISSUE TITLE
💮stampino: a Nimib based cookie cutter

### ISSUE THREAD

**Issue by pietroppeter** [pietroppeter/nimib#115]
having done twice (nimib, nimibook) the operation of adding PR deploy preview with Netflix, and counting that it will be useful for other projects (I will likely need to do it at least once more within the month), I plan to create as a nimib side project **stampino**.

The idea is to have a nimib based [cookie cutter](https://github.com/cookiecutter/cookiecutter) ("stampino" is a possible italian translation of cookie cutter). Not that I actually know very much about cookie-cutter itself (used it twice years ago), but the idea is to have a way to add folders and documents starting from templates and it seem that does just that (although cookie-cutter seems to focus on projects). First target would be to add some common GitHub actions for nimib based projects but it could be used in more ways than that.

As an initial design idea, stampino would come with a set of predefined "stampini" (set of templates), initially some common nim/nimib actions, but in principle it could be extended by pointing to a given folder or repo or folder in a repo.
Context content would initially come from a `[stampino]` section in `nimib.toml`.

It would also be an experiment in using nimib for a different use case than producing documents.

💮 is also known as well done stamp: https://www.floom.com/us/magazine/article/world-emoji-day

**Comment by HugoGranstrom** [comment]
> I will likely need to do it at least once more within the month

Smells like a new nimib theme is coming soon :open_mouth: 

I really like the idea of having a template to make new nimib themes, but I'm not really following how you will use nimib for this. It feels like this is just some `makedir` + `writeFile` action going on to create the files, nothing that nimib itself is needed for.

**Comment by pietroppeter** [comment]
> Smells like a new nimib theme is coming soon 😮

yes, the idea to work on #77 (once I have completed ongoing stuff)

> I really like the idea of having a template to make new nimib themes, but I'm not really following how you will use nimib for this. It feels like this is just some makedir + writeFile action going on to create the files, nothing that nimib itself is needed for.

yes, it is probably not much more than that, but you will also need to substitute some names (netlify site name) or parts of document so some templating could be useful. It might indeed be that nimib is not a good fit and I am wasting a good name and emoji for that... :)

**Comment by HugoGranstrom** [comment]
> A binary

Ok :+1: I've come around and making use of nimib's ability to understand the structure of a nimib project does make sense to facilitate. And in theory `nbFile` could be used to create the needed files...
